Offer test vectors

thomash-acinq opened the pull request:
- Add test vectors from https://github.com/lightning/bolts/pull/798
- Update the TLV ranges for offers and invoice requests

t-bast commented:
Should we update this with the latest test vectors from the Bolt 12 spec PR @thomash-acinq? Are the spec test vectors up-to-date or are we waiting for changes there?

thomash-acinq commented:
I've updated the test vectors and the TLV ranges for offers.
The reason why this was this was still a draft is that there are two cases where I disagree with the spec test vectors:
- 22 is a known feature and an offer with this feature is valid
- amount and description are optional

I've modified the test vectors for these two cases but it should be updated in the spec.

t-bast reviewed (approved):
> The reason why this was this was still a draft is that there are two cases where I disagree with the spec test vectors:

Gotcha, I'm not sure about the feature bit one, but the other one I agree with you. Can you make sure this is discussed on the spec PR and Rusty is aware of it, so that we can converge? We'll need everyone to validate the test vectors to be able to merge the spec PR.
